packages/react/src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import IconButton from '../IconButton';
+import TooltipTabstop from '../TooltipTabstop';
+import Icon from '../Icon';
+
+type PaginationVariant = 'dark';
+
+interface Props {
+  totalItems: number;
+  itemsPerPage?: number;
+  currentPage?: number;
+  statusLabel?: React.ReactElement;
+  firstPageLabel?: string;
+  previousPageLabel?: string;
+  nextPageLabel?: string;
+  lastPageLabel?: string;
+  onNextPageClick?: () => void;
+  onPreviousPageClick?: () => void;
+  onFirstPageClick?: () => void;
+  onLastPageClick?: () => void;
+  className?: string;
+  variant?: PaginationVariant;
+}
+
+const Pagination = React.forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      totalItems,
+      itemsPerPage = 10,
+      currentPage = 1,
+      statusLabel,
+      firstPageLabel = 'First page',
+      previousPageLabel = 'Previous page',
+      nextPageLabel = 'Next page',
+      lastPageLabel = 'Last page',
+      onNextPageClick,
+      onPreviousPageClick,
+      onFirstPageClick,
+      onLastPageClick,
+      className,
+      variant,
+      ...other
+    },
+    ref
+  ) => {
+    const itemStart = currentPage * itemsPerPage - itemsPerPage + 1;
+    const itemEnd = Math.min(itemStart + itemsPerPage - 1, totalItems);
+    const isLastPage = itemEnd === totalItems;
+
+    return (
+      <div
```

Comment:
Should this be `nav` or `role="navigation"`?

Reply:
🤷 I originally coded it as a `<nav />` but felt kinda funky considering all the controls are buttons. I think having it as a list conveys everything we need but am more than happy to change this back to a `<nav />`.

Reply:
I'm not picky here as we can slate this for a future improvement™ per comment https://github.com/dequelabs/cauldron/pull/485#discussion_r775977006